src/main/scala/ai/privado/entrypoint/ScanProcessor.scala

```diff
@@ -332,19 +332,19 @@ object ScanProcessor extends CommandProcessor {
     * @return
     *   processed rules
     */
-  def getProcessedRule(lang: Set[Language]): RuleCache = {
-    AppCache.repoLanguage =
+  def getProcessedRule(lang: Set[Language], appCache: AppCache): RuleCache = {
+    appCache.repoLanguage =
       lang.head // we are caching the repo language here, and we will use this to get the repo's lang
     val ruleCache      = new RuleCache()
-    val processedRules = processRules(lang, ruleCache)
+    val processedRules = processRules(lang, ruleCache, appCache)
     ruleCache.setRule(processedRules)
     ruleCache
   }
 
-  private def processCpg() = {
+  private def processCpg(appCache: AppCache) = {
     val sourceRepoLocation = File(config.sourceLocation.head).path.toAbsolutePath.toString.stripSuffix("/")
     // Setting up the application cache
-    AppCache.init(sourceRepoLocation)
+    // val appCache = new AppCache(sourceRepoLocation)
```

Comment:
where are we setting `sourceRepoLocation`?

Reply:
yes, there only https://github.com/Privado-Inc/privado-core/pull/1023/files#diff-47c8acbc2ffafd950e96bdaf4e0b8d26671f59626a039824e12b051932d422ebR347.